python/src/gluon_ir.cc

```diff
@@ -299,6 +299,11 @@ void init_gluon_ir(py::module &&m) {
              self.create<ttng::AsyncTMAScatterOp>(descPtr, xOffsets, yOffset,
                                                   src);
            })
+      .def("create_fence_async_shared",
+           [](GluonOpBuilder &self, bool bCluster) -> OpState {
+             return self.create<ttng::FenceAsyncSharedOp>(bCluster);
```

Comment:
IIRC @ThomasRaoux thought we could run the fence insertion pass and avoid the need to manually write fences. Any thoughts?

Reply:
The only thing FenceInsertion does is look for `mma(local_alloc(a), b, d)` and place a `ttng.fence_async_shared` before the MMA. Other cases requires manual fence insertion, and these cases are now exposed to the user :/

I think manual fence insertion is a huge footgun (cc @pawelszczerbuk) but due to non-zero performance cost we can't just bake it into the API. I think we should look into making FenceInsertion more sophisticated or have a pass to removes unnecessary fences.